specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/webapplicationfirewall.json

```diff
@@ -667,25 +675,109 @@
           }
         },
         "selectorMatchOperator": {
-          "type": "string",
-          "enum": [
-            "Equals",
-            "Contains",
-            "StartsWith",
-            "EndsWith",
-            "EqualsAny"
-          ],
           "description": "When matchVariable is a collection, operate on the selector to specify which elements in the collection this exclusion applies to.",
-          "x-ms-enum": {
-            "name": "OwaspCrsExclusionEntrySelectorMatchOperator",
-            "modelAsString": true
-          }
+          "$ref": "#/definitions/MatchOperator"
         },
         "selector": {
           "type": "string",
           "description": "When matchVariable is a collection, operator used to specify which elements in the collection this exclusion applies to."
         }
       }
+    },
+    "MatchOperator": {
+      "type": "string",
+      "enum": [
+        "Equals",
+        "Contains",
+        "StartsWith",
+        "EndsWith",
+        "EqualsAny"
+      ],
+      "description": "Defines the logic for comparing two strings",
+      "x-ms-enum": {
+        "name": "OwaspCrsExclusionEntrySelectorMatchOperator",
+        "modelAsString": true
+      }
+    },
+    "BotDetection": {
+      "description": "Allows the detection of bots and specify action for each bot group",
+      "required": [
+        "good", 
+        "neutral", 
+        "bad"
+      ],
+      "properties": {
+        "good": {
+          "description": "Describes the action to be taken when detecting a good bot.",
+          "$ref": "#/definitions/BotDetection"
+        },
+        "neutral": {
+          "description": "Describes the action to be taken when detecting a neutral bot.",
+          "$ref": "#/definitions/BotDetection"
+        },
+        "bad": {
+          "description": "Describes the action to be taken when detecting a bad bot.",
+          "$ref": "#/definitions/BotDetection"
+        },
+        "overrides": {
+          "description": "List of custom bot categories which take precedence over the predefined categories",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CustomBotCategory"
+          }
+        }
+      }
+    },
+    "BotIdentifier": {
+      "description": "Defines a way to capture a bot request",
+      "required": [
+        "type",
+        "value",
+        "matchOperator"
+      ],
+      "properties": {
+        "type": {
+          "description": "Describes which property of the request to match against",
+          "type": "string",
+          "enum": [
+            "IP",
+            "MAC Address"
+          ]
+        },
+        "value": {
+          "description": "Describes the value against which property of the request will be matched",
+          "type": "string"
+        },
+        "matchOperator": {
```

Comment:
I wasn't sure if this was necessary, but I recall Avinash mentioning support for operators, so I added it here

Reply:
update: Checked with eric and the match operators are part of requirements